src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -99,6 +115,26 @@ public void Reset()
             ResponseBody = null;
             DuplexStream = null;
 
+            var httpConnectionFeature = this as IHttpConnectionFeature;
+            if (httpConnectionFeature != null)
+            {
+                if (_remoteEndPoint != null)
+                {
+                    httpConnectionFeature.RemoteIpAddress = _remoteEndPoint.Address;
```

Comment:
these can be simplified to `httpConnectionFeature.RemoteIpAddress = _remoteEndPoint?.Address;` since you always want to reset the feature.